common/persistence/persistenceTestBase.go

```diff
@@ -77,6 +77,7 @@ type (
 		TaskMgr              TaskManager
 		HistoryMgr           HistoryManager
 		MetadataManager      MetadataManager
+		MetadataManagerV2    MetadataManager
```

Comment:
Can we just have these 2 versions only at the persistence layer?  I'm a little worried bubbling it up across all layers of application might cause problems and bugs.  There are just too many changes because of that.

Reply:
you see, i am using the same interface.
so, anywhere other than persistence (and domain frontend API and the domain cache) will not use the new metadata manager